cpp/src/plasma/store.cc

```diff
@@ -532,7 +532,15 @@ int PlasmaStore::RemoveFromClientObjectIds(const ObjectID& object_id,
 
 void PlasmaStore::EraseFromObjectTable(const ObjectID& object_id) {
   auto& object = store_info_.objects[object_id];
-  PlasmaAllocator::Free(object->pointer, object->data_size + object->metadata_size);
+  if (object->device_num == 0) {
+    PlasmaAllocator::Free(object->pointer, object->data_size + object->metadata_size);
+  } else {
+#ifdef PLASMA_CUDA
+    std::shared_ptr<CudaContext> context_;
+    manager_->GetContext(object->device_num - 1, &context_);
+    context_->Free(object->pointer, object->data_size + object->metadata_size);
```

Comment:
Unchecked Status here (this would fail CI if we had GPU-enabled builds). Seems that `EraseFromObjectTable` will need to return Status now?

Reply:
Status is checked.